internal/app/machined/pkg/runtime/v1alpha1/platform/gcp/gcp.go

```diff
@@ -60,7 +62,10 @@ func (g *GCP) ExternalIPs() (addrs []net.IP, err error) {
 		resp *http.Response
 	)
 
-	if req, err = http.NewRequest("GET", GCExternalIPEndpoint, nil); err != nil {
+	ctx, ctxCancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer ctxCancel()
+
+	if req, err = http.NewRequestWithContext(ctx, "GET", GCExternalIPEndpoint, nil); err != nil {
```

Comment:
nit: There is `http.MethodGet` that we could use instead of `"GET"`.

Reply:
yep, I used that it many places, not sure how I missed this one!

proper fix is to add context to all those platform methods, but this should come as another PR